src/research-study.ts

```diff
@@ -158,216 +101,28 @@ export class ResearchStudy {
   sponsor?: Reference;
   principalInvestigator?: Reference;
   site?: Reference[];
-  contained?: (Group | Location | Organization | Practitioner)[];
-
-  constructor(trial: TrialObject, id: number) {
-    this.id = String(id);
-    if (trial.nctId) {
-      this.identifier = [{use: "official", system: "http://clinicaltrials.gov", value: trial.nctId}];
-    }
-    if (trial.title) {
-      this.title = trial.title;
-    }
-    if (trial.trialStatus) {
-      this.status = trial.trialStatus
-    }
-    if (trial.phase) {
-      this.phase = {coding: [{system: "http://terminology.hl7.org/CodeSystem/research-study-phase", code: trial.phase, display: trial.phase}], text: trial.phase}; //may need to be fixed
-    }
-    if (trial.studyType) {
-      this.category = [{text: trial.studyType}];
-    }
-    if (trial.conditions != "[]") {
-      this.condition = this.convertStringArrayToCodeableConcept(trial.conditions);
-    }
-    if (trial.contactInfo) {
-      this.contact = this.setContact(trial.contactInfo);
-    }
-    if (trial.keywords != "[]") {
-      this.keyword = this.convertStringArrayToCodeableConcept(trial.keywords);
-    }
-    if (trial.countries != "[]") {
-      this.location = this.convertStringArrayToCodeableConcept(trial.countries);
-    }
-    if (trial.detailedDescription) {
-      this.description = trial.detailedDescription;
-    }
-    if (typeof trial.armGroups[Symbol.iterator] === 'function') { // ts returns {} when empty, which is not iterable
-      this.arm = this.setArm(trial.armGroups);
-    }
-    if (trial.objective) {
-      this.objective = [{name: trial.objective}];
-    }
-    if (trial.criteria) {
-      this.enrollment = [{reference: "#group" + this.id, type: "Group", display: trial.criteria}];
-    }
-    if (trial.sponsor) {
-      this.sponsor = {reference: "#org" + this.id, type: "Organization"};
-    }
-    if (trial.overallOfficialName) {
-      this.principalInvestigator = {reference: "#practitioner" + this.id, type: "Practitioner"};
-    }
-    if (trial.sites != []) {
-      this.site = this.setSiteReferences(trial.sites);
-    }
-    //Checks if research study contains enrollment criteria 
-
-    if(!trial.criteria){
-      this.enrollment = [{ reference: `#group${this.id}`, type: "Group", display:  this.addCriteria() }];
-     
-    }
-    if(!trial.detailedDescription){
-      this.description=  this.addSummary();
-
-    }
-    /** Commented out for the time being since convertPhaseCode is 
-    if(!trial.phase){
-      this.phase = {coding: [{system: "http://terminology.hl7.org/CodeSystem/research-study-phase", code: this.convertPhaseCode(this.addPhase()), display: this.addPhase()}], text: this.addPhase()};
-     
-    }
+  contained?: (Group | Location | Organization | Practitioner)[]; //
+
+  constructor(trial, id: number) {
```

Comment:
It might be confusing to set all these parameters fhir compliant/easy to do incorrectly, could be helpful to include an example of sorts

Reply:
I added a link the trialscope implementation in the comments to give an example of a filled out ResearchStudy constructor. I think that should provide enough guidance.